### DISCUSSION TITLE
Install en_core_web_trf on Windows get conflicting dependencies error

### DISCUSSION THREAD

**Discussion by xxyzz** [explosion/spaCy#11891] [Help: Installation] [answered]
## How to reproduce the behaviour

Run this command `py -m pip install https://github.com/explosion/spacy-models/releases/download/en_core_web_trf-3.4.1/en_core_web_trf-3.4.1-py3-none-any.whl` will get the following error from pip:

```
ERROR: Cannot install en-core-web-trf because these package versions have conflicting dependencies.

The conflict is caused by:
    spacy-transformers 1.1.8 depends on torch>=1.6.0
    spacy-transformers 1.1.7 depends on torch>=1.6.0
    spacy-transformers 1.1.6 depends on torch>=1.6.0
    spacy-transformers 1.1.5 depends on torch>=1.6.0
    spacy-transformers 1.1.4 depends on torch>=1.6.0
ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
    spacy-transformers 1.1.3 depends on torch>=1.6.0
    spacy-transformers 1.1.2 depends on torch>=1.6.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict
```

The full log can be found at here: https://github.com/xxyzz/WordDumb/actions/runs/3573895840/jobs/6008499300

I also tried the `python -m spacy download en_core_web_trf` command but got the same error.

## Your Environment

* Operating System: Microsoft Windows Server 2022
* Python Version Used: 3.11.0
* spaCy Version Used: 3.4.3
* Environment Information:

**Comment by adrianeboyd** [comment, marked as answer]
It looks like torch doesn't fully support python 3.11 yet. I would suggest using python 3.10 instead for now.

**Reply by xxyzz** [reply]
I just found torch only provides a single 3.11 wheel file for Linux and was about to close the issue. Thanks for your help!

**Comment by xxyzz** [comment]
Now I get another problem, the torch package for Windows on pypi.org is the CPU build version. This would get `AssertionError: Torch not compiled with CUDA enabled` error from torch when spaCy loading the trf model. The CUDA build torch package should be installed with this command `python -m pip install torch --extra-index-url https://download.pytorch.org/whl/cu117`.

**Reply by polm** [reply]
Glad you figured it out. Torch installs can be a little tricky, it's best to use their [install widget](https://pytorch.org/) to get it set up.
